src/gevent/libev/corecext.pyx

```diff
@@ -203,32 +202,31 @@ cpdef _flags_to_list(unsigned int flags):
     return result
 
 
-
-basestring = (bytes, str)
-
-
-
-cpdef unsigned int _flags_to_int(object flags) except? -1:
+cpdef unsigned int _flags_to_int(object flags) except *:
```

Comment:
`except? -1` (or whatever integer value you chose) is still more efficient than `except *`. The latter must always ask CPython if something happened, whereas the first only needs to do that if the functions really returned exactly the exception value. I'd recommend sticking with some unlikely exception value here, even arbitrarily chosen. `except *` is mostly useful for `void` or `struct` returns.

Reply:
Thanks. `except *` was one of the things I tried to workaround the code generation issue. I decided to keep it because this isn't performance sensitive code, and there's something vaguely unsettling about comparing an `unsigned int` to -1.